From 3040ec8468ed911c1f122dad292d04e5a8d77457 Mon Sep 17 00:00:00 2001
From: Nadav Hugim <nhugim@habana.ai>
Date: Wed, 05 Jan 2022 16:07:05 +0200
Subject: [PATCH] [SW-66652] fake security: fix stack overflow

moving out struct hl_fit_sec_desc outside  of function

Change-Id: I27ac01c27356cae3e217553d9b478d3e029995a7
Signed-off-by: Nadav Hugim <nhugim@habana.ai>
---

diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index 24f520c..239b4fe 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -122,6 +122,7 @@
 };
 
 #ifdef FAKE_SECURITY
+static struct hl_fit_sec_desc fit_sec;
 static u8 hash_value[SHA256_SUM_LEN];
 #define FAKE_OTP_OFFSET	(2 * 1024 * 1024)
 static void preboot_security_copy_otp(struct hl_sec_otp *otp)
@@ -169,7 +170,6 @@
 static void save_hash_pub_key(const void *img, size_t img_size)
 {
 	const void *fit_sec_off;
-	struct hl_fit_sec_desc fit_sec;
 
 	/* hl fit files are padded so the size is 4 bytes aligned */
 	img_size = ROUND_UP4(img_size);
